src/js/shared/Events/index.mjs

```diff
@@ -64,7 +72,9 @@ const doListen = function(module, event, callback, once) {
 
     let events = (event === '*' ? validEvents[module] : [event]) // explodes wildcards into an array
     let promises = []
-    const key = module + '.' + event // this might be a wildcard, e.g. 'lifecycle.*'
+    const hasContext = Object.values(context).length > 0
+    const contextKey = Object.keys(context).sort().map(key => key + '=' + JSON.stringify(context[key])).join('&')
```

Comment:
I don't quite understand how this works.
The callback is stored now using the contextKey, but when emitting the event, I don't think it uses this same key to find the callback?

Reply:
You might be right. let me look closer at my tests / code.

Reply:
Yeah, it was totally broken. I had to refactor the Event/Transport integration, but it's much more solid now